### PR TITLE
Long requests throttled for long time

### PR DESCRIPTION
**Pull request by wojtek-t**
See: https://github.com/kubernetes/kubernetes/issues/19048#issuecomment-184812462 for more details

### CODE REVIEW

File: pkg/client/unversioned/request.go [hunk 1]
```diff
@@ -683,9 +689,13 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 		return nil, r.err
 	}
 
+	now := time.Now()
 	if r.throttle != nil {
 		r.throttle.Accept()
 	}
+	if latency := time.Since(now); latency > longThrottleLatency {
+		glog.Warningf("Throttling request took %v, request: %s", latency, r.URL().String())
+	}
```
Comment (lavalamp) [inline pkg/client/unversioned/request.go]: Perhaps replace these 7 lines with a single function, r.tryThrottle or some such, to reduce code duplication? Also you could put the timing code in side the if statement, but that's a silly nit.

**Comment by wojtek-t** [comment]
@lavalamp - PTAL

File: pkg/client/unversioned/request.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/client/unversioned/request.go]: done

**Comment by lavalamp** [comment]
LGTM, thanks!

**Pull request merged**